pdc/apps/partners/serializers.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Red Hat
+# Licensed under The MIT License (MIT)
+# http://opensource.org/licenses/MIT
+#
+from rest_framework import serializers
+
+from pdc.apps.common.fields import ChoiceSlugField
+from pdc.apps.release import models as release_models
+from . import models
+
+
+class PartnerTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.PartnerType
+        fields = ('name', )
+
+
+class PartnerSerializer(serializers.ModelSerializer):
+    type = ChoiceSlugField(slug_field='name',
+                           queryset=models.PartnerType.objects.all())
+
+    class Meta:
+        model = models.Partner
+        fields = ('short', 'name', 'binary', 'source', 'type', 'enabled',
+                  'ftp_dir', 'rsync_dir')
+
+
+class PartnerMappingSerializer(serializers.ModelSerializer):
+    partner = serializers.SlugRelatedField(slug_field='short',
+                                           queryset=models.Partner.objects.all())
+    release = serializers.CharField(source='variant_arch.variant.release.release_id')
+    variant = serializers.CharField(source='variant_arch.variant.variant_uid')
+    arch = serializers.CharField(source='variant_arch.arch.name')
+
+    class Meta:
+        model = models.PartnerMapping
+        fields = ('partner', 'release', 'variant', 'arch')
+
+    def validate(self, attrs):
+        try:
+            variant_arch = attrs.get('variant_arch', {})
+            release_id   = variant_arch.get('variant', {}).get('release', {}).get('release_id', '')
+            variant_uid  = variant_arch.get('variant', {}).get('variant_uid', '')
+            arch         = variant_arch.get('arch', {}).get('name', '')
+            attrs['variant_arch'] = release_models.VariantArch.objects.get(
+                variant__release__release_id=release_id,
+                variant__variant_uid=variant_uid,
+                arch__name=arch
+            )
+        except release_models.VariantArch.DoesNotExist:
+            raise serializers.ValidationError(
+                'No VariantArch for release_id=%s, variant_uid=%s, arch=%s'
+                % (release_id, variant_uid, arch)
+            )
+        try:
+            models.PartnerMapping.objects.get(**attrs)
```

Comment:
why not using `self.instance` to check if partners mapping already exists? since both `create` and `update` validation will run into this function.